ui/src/style/theme/_buttons.scss

```diff
@@ -0,0 +1,433 @@
+    $bg-color,
+    $bg-color-hover,
+    $bg-color-press,
+    $text-color,
+    $text-color-hover
+  ) {
+
+  // Default State
+  background-color: $bg-color;
+  color: $text-color;
+  text-shadow: none;
+  border-width: 0;
+
+  // Focus State
+  &:focus {
+    background-color: $bg-color;
+    color: $text-color;
+    box-shadow: none;
+  }
+
+  // Hover State
+  &:hover,
+  &:focus:hover {
+    background-color: $bg-color-hover;
+    color: $text-color-hover;
+    cursor: pointer;
+  }
+  // MouseDown State
+  &.active,
+  &.active:hover,
+  &:active,
+  &:active:hover,
+  &:focus:active,
+  &:focus:active:hover,
+  .dropdown.open &.dropdown-toggle  {
+    background-color: $bg-color-press;
+    color: $text-color-hover;
+    cursor: pointer;
+  }
+  // Disabled State
+  &.disabled,
+  &[disabled],
+  fieldset[disabled] & {
+    opacity: 1;
+    font-style: italic;
+    background-color: $g5-pepper;
+    color: $g7-graphite;
+    box-shadow: none;
+
+    &:hover,
+    &:active,
+    &:focus,
+    &.active,
+    &:active:focus {
+      opacity: 1;
+      background-color: $g5-pepper;
+      color: $g7-graphite;
+      cursor: not-allowed;
+      box-shadow: none;
+    }
+    &:after {
+      display: none;
+    }
+  }
+}
+
+a.btn-default,
+div.btn-default,
+button.btn-default,
+input.btn-default {
+  @include btn-base-styles(
+    $g5-pepper,
+    $g6-smoke,
+    $g7-graphite,
+    $g14-chromium,
+    $g18-cloud
+  );
+}
+a.btn-primary,
+div.btn-primary,
+button.btn-primary,
+input.btn-primary {
+  @include btn-base-styles(
+    $c-pool,
+    $c-laser,
+    $c-hydrogen,
+    $g20-white,
+    $g20-white
+  );
+}
+a.btn-success,
+div.btn-success,
+button.btn-success,
+input.btn-success {
+  @include btn-base-styles(
+    $c-rainforest,
+    $c-honeydew,
+    $c-krypton,
+    $g20-white,
+    $g20-white
+  );
+}
+a.btn-info,
+div.btn-info,
+button.btn-info,
+input.btn-info {
+  @include btn-base-styles(
+    $g7-graphite,
+    $g8-storm,
+    $g9-mountain,
+    $g16-pearl,
+    $g20-white
+  );
+}
+a.btn-warning,
+div.btn-warning,
+button.btn-warning,
+input.btn-warning {
+  @include btn-base-styles(
+    $c-star,
+    $c-comet,
+    $c-potassium,
+    $g20-white,
+    $g20-white
+  );
+}
+a.btn-danger,
+div.btn-danger,
+button.btn-danger,
+input.btn-danger {
+  @include btn-base-styles(
+    $c-curacao,
+    $c-dreamsicle,
+    $c-tungsten,
+    $g20-white,
+    $g20-white
+  );
+}
+a.btn-alert,
+div.btn-alert,
+button.btn-alert,
+input.btn-alert {
+  @include btn-base-styles(
+    $c-pineapple,
+    $c-thunder,
+    $c-sulfur,
+    $g20-white,
+    $g20-white
+  );
+}
+
+
+// Outline Button styles mixin
+// -----------------------------------------------------------------------------
+@mixin btn-outline-styles (
+    $text-color,
+    $text-color-hover,
+    $border-color,
+    $border-color-hover,
+    $border-color-press
+  ) {
+
+  // Adjusting line height in response to more borders
+  line-height: ($form-md-height - 4px);
```

Comment:
Do we need calcs here?

Reply:
Might not need them, had them in there to be cautious. I think the compiler should know to evaluate the expression without the explicit `()` around it